### DISCUSSION TITLE
BackupStorageLocation (BSL) configuration requirements for non-admin backup/restore operations

### DISCUSSION THREAD

**Discussion by shubham-pampattiwar** [migtools/oadp-non-admin#28] [General]
We need to address the following questions:
- Can BSL be a cluster-admin responsibility ?
- Is non-admin user's read access to BSL acceptable ?

**Comment by dankenigsberg** [comment]
I cannot vote here as I don't know what BackupStorageLocation is.

**Reply by mpryc** [reply]
@dankenigsberg On the upstream Velero documentation page there is a nice explanation of what `BackupStorageLocation` is: https://velero.io/docs/main/locations/#overview

The discussion here is if the user should have option to define such BSL and use those in the backups or the user should solely relay on the administrator specifying those places.

In other words this discussion is around the allowance or possibility for the non admin user to specify where the backups are being stored or it should be responsibility of an cluster administrator to create such palces and let non admin user decide which from those defined by administrator places use or the default one if the user won't have knowledge around the BSL location.

Note that each BSL requires some sort of credentials to be specified such as Access or Secret keys.

**Reply by dankenigsberg** [reply]
As a tenant of a cluster, I would like to have full control of when I create, delete, backup and restore my VMs. I'd like to decide were I'm backing them up and with which credentials. It'd like to provide my own S3 bucket. If I want to create a backup every hour, I don't expect my cluster-admin to let me consume storage from their cluster-wide bucket.

**Comment by shubham-pampattiwar** [comment]
Issue related to this discussion: https://github.com/migtools/oadp-non-admin/issues/36
